substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -469,12 +469,59 @@ where
 		self.view_store.futures_at(at)
 	}
 
+	/// Searches in the view store for a view by iterating through the fork of
+	/// the `at` block, up to the last finalized block.
+	///
+	/// Returns with a pair of a maybe view and a set of enacted blocks (possibly empty if no view
+	/// is found), usually starting at the last known block of the fork, up to the block
+	/// corresponding to the found view.
+	fn find_best_view(
+		&self,
+		at: &HashAndNumber<Block>,
+	) -> (Option<Arc<View<ChainApi>>>, Vec<Block::Hash>) {
+		let mut enacted_blocks = Vec::new();
+		let mut best_view = None;
+
+		let mut at_hash = at.hash;
+		let mut at_number = at.number;
+
+		let last_finalized = self.enactment_state.lock().recent_finalized_block();
+		let Some(last_finalized_number) = self.api.resolve_block_number(last_finalized).ok() else {
+			return (None, Vec::new());
```

Comment:
```suggestion
			return None;
```

Reply:
I moved this logic up to `ready_at_light` and returned an empty tx set for it. Decided to treat this in the same way we treat not being able to resolve the `at` hash to a block number. WDYT?

[9d47e15](https://github.com/paritytech/polkadot-sdk/pull/8533/commits/9d47e15417b4cf3fe54ec942fc883f852e494b32)